pkg/kustomize/kustomize_cli.go

```diff
@@ -0,0 +1,34 @@
+package kustomize
+
+import (
+	"github.com/jenkins-x/jx/pkg/util"
+)
+
+// KustomizeCLI implements common kustomize actions based on kustomize CLI
+type KustomizeCLI struct {
+	Runner util.Commander
+}
+
+// NewKustomizeCLI creates a new KustomizeCLI instance configured to use the provided kustomize CLI in
+// the given current working directory
+func NewKustomizeCLI() *KustomizeCLI {
+	runner := &util.Command{
+		Name: "kustomize",
+	}
+	cli := &KustomizeCLI{
+		Runner: runner,
+	}
+	return cli
+}
+
+// Version executes the Kustomize version command and returns its output
+func (k *KustomizeCLI) Version(extraArgs ...string) (string, error) {
+	args := []string{"version", "--short"}
+	args = append(args, extraArgs...)
```

Comment:
I am wondering whether we should place this whole regular expression stuff to extract the atual semantic version into here. This function would eg just return 3.5.4. Otherwise, we have the logic spread across mutliple places and it is not clear why you, for example, use a regular expression later to extract the actual version.